test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJavaAPI.java

```diff
@@ -106,9 +168,28 @@ public static void main(String args[]) {
 		    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found");
 		    }
 		    newCacheCount = cacheList.size();
-		    if ((newCacheCount == -1) || (newCacheCount != (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount))) {
-		    	fail("SharedClassUtilities.getSharedCacheInfo failed: Invalid number of cache found\t" +
-		    			"expected: " + (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount) + "\tfound: " + newCacheCount);
+		    
+		    if (dirGroupAccess != null) {
+		    	cacheGroupAccessList = SharedClassUtilities.getSharedCacheInfo(dirGroupAccess, SharedClassUtilities.NO_FLAGS, false);
+		    	if (cacheGroupAccessList == null) {
+			    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found");
```

Comment:
Better include dirGroupAccess in the error message. Similarly for the following cases

Reply:
Updated.